core/field.ts

```diff
@@ -704,45 +686,32 @@ export abstract class Field implements IASTNodeLocationSvg,
     const constants = this.getConstants();
```

Comment:
In the case of `getConstants()`, I actually think it's preferable *not* to throw an error since the return type properly informs the user that it could be null. (Unless an error needs to be thrown for it being something other than `ConstantProvider` or `null`?)

Reply:
Makes sense, sounds good =)